src/pipeline.js

```diff
@@ -109,34 +105,38 @@ class Pipeline {
       if (ext && ext.after && typeof ext.after === 'object') {
         Object.keys(ext.after).map((key) => this.attach.after(key, ext.after[key]));
       }
+      if (ext && ext.replace && typeof ext.replace === 'object') {
+        Object.keys(ext.replace).map((key) => this.attach.replace(key, ext.replace[key]));
+      }
       this.sealed = true;
     };
 
     /**
      * Registers an extension to the pipeline.
      * @param {String} name - name of the extension point (typically the function name).
      * @param {pipelineFunction} f - a new pipeline step that will be injected relative to `name`.
-     * @param {integer} before - where to insert the new function (true = before, false = after)
+     * @param {integer} offset - where to insert the new function (-1: replace, 0: before, 1: after)
      */
-    const attachGeneric = (name, f, before) => {
-      const offset = before ? 0 : 1;
+    const attachGeneric = (name, f, offset) => {
       // find the index of the function where the resolved ext name
-      // matches the provided name by searching the list of pre and
-      // post functions
-      const foundpres = this._pres
-        .findIndex((pre) => pre && pre.ext && pre.ext === name);
-      const foundposts = this._posts
-        .findIndex((post) => post && post.ext && post.ext === name);
-
-      // if something has been found in either lists, insert the
+      // matches the provided name by searching the list of step functions
+      const foundstep = this._steps
+        .findIndex((step) => step && step.ext && step.ext === name);
+
+      // if something has been found in the list, insert the
       // new function into the list, with the correct offset
-      if (foundpres !== -1) {
-        this._pres.splice(foundpres + offset, 0, f);
-      }
-      if (foundposts !== -1) {
-        this._posts.splice(foundposts + offset, 0, f);
-      }
-      if (foundpres === -1 && foundposts === -1) {
+      if (foundstep !== -1) {
+        if (offset < 0) {
```

Comment:
this is counter intuitive: `<0` should be before, `==0` replace, and `>0` after.